### PR TITLE
Use ninja-build in CI. Use cmake toolchains.

### PR DESCRIPTION
**Pull request by snellers**

### CODE REVIEW

File: .github/workflows/c-cpp.yml [hunk 1]
```diff
@@ -22,9 +22,9 @@ jobs:
     steps:
       - uses: actions/checkout@v1
       - name: install-compiler
```
Comment (snellers) [inline .github/workflows/c-cpp.yml:24]: For the time being, no use of toolchains in the CI.

File: Makefile [hunk 2]
```diff
@@ -5,6 +5,12 @@ default: install
 help: # with thanks to Ben Rady
 	@grep -E '^[0-9a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
 
+ifeq ($(shell which g++-11),)
```
Comment (snellers) [inline Makefile:8]: Axed the CC/CXX business. Now just detect g++-11 and fallback to 10. But also allow someone to pass `TOOLCHAIN` as an arg to `make` if they really want to.

File: Makefile [hunk 3]
```diff
@@ -98,10 +99,8 @@ stop: dirs  ## Stop Xania
 .PHONY: restart
 restart: stop start  ## Restart Xania
 
-# Grr older cmakes don't support -S -B
 $(BUILD_ROOT)/CMakeCache.txt:
-	@mkdir -p $(BUILD_ROOT)
-	cd $(BUILD_ROOT) && $(CMAKE) .. $(CMAKE_GENERATOR_FLAGS) -DCMAKE_BUILD_TYPE=$(BUILD_TYPE) -DCMAKE_INSTALL_PREFIX=$(INSTALL_DIR)
+	$(CMAKE) -S . -B $(BUILD_ROOT) $(CMAKE_GENERATOR_FLAGS) --toolchain toolchain/$(TOOLCHAIN).cmake -DCMAKE_BUILD_TYPE=$(BUILD_TYPE) -DCMAKE_INSTALL_PREFIX=$(INSTALL_DIR)
```
Comment (snellers) [inline Makefile:103]: We stipulate a recent cmake now so using -S and -B again

Comment (snellers) [reply Makefile:103]: --toolchain is tidier than `DCMAKE_TOOLCHAIN_FILE`

File: .github/workflows/c-cpp.yml [hunk 4]
```diff
@@ -38,10 +38,10 @@ jobs:
...
-        run : sudo apt-get update && sudo apt-get install g++-11 -y
+        run : sudo apt-get update && sudo apt-get install ninja-build g++-11 -y
       - name: configure
         run: |
-          cmake -DCMAKE_BUILD_TYPE=Release \
+          cmake -G Ninja -DCMAKE_BUILD_TYPE=Release \
```
Comment (mattgodbolt) [inline .github/workflows/c-cpp.yml:44]: Love it

File: README.md [hunk 5]
```diff
@@ -25,16 +25,23 @@ $ sudo apt install build-essential manpages-dev software-properties-common
 $ sudo add-apt-repository ppa:ubuntu-toolchain-r/test
 $ sudo apt update && sudo apt install gcc-11 g++-11
 
+# Optional step: We recommend installing ninja-build as it speeds the builds up significantly.
```
Comment (mattgodbolt) [inline README.md:28]: Does this speed the builds up? Maybe a little bit? (Will test actually!)

**Review by mattgodbolt — APPROVED** [review]

**Pull request merged**